src/main/java/com/aws/greengrass/util/orchestration/WinswUtils.java

```diff
@@ -74,13 +81,19 @@ public boolean setupSystemService(KernelAlternatives kernelAlternatives, Nucleus
         return false;
     }
 
-    private void interpolateServiceTemplate(Path src, Path dst, KernelAlternatives kernelAlternatives)
-            throws IOException {
-        try (BufferedReader r = Files.newBufferedReader(src); BufferedWriter w = Files.newBufferedWriter(dst)) {
+    private void interpolateServiceTemplate(
+            Path src, Path dst, KernelAlternatives kernelAlternatives) throws IOException {
+        try (BufferedReader r = Files.newBufferedReader(src);
+                BufferedWriter w = Files.newBufferedWriter(dst)) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Path traversal vulnerability detected. User-controlled input in file paths
can allow attackers to access files outside intended directories using
`../` sequences. Secure your code by using framework functions like
`getCanonicalPath()`, `normalize()`, or validating paths with
`.startsWith()` checks. Learn more:
https://cwe.mitre.org/data/definitions/22.html